src/node_sqlite.cc

```diff
@@ -1763,21 +1763,25 @@ void DatabaseSync::ApplyChangeset(const FunctionCallbackInfo<Value>& args) {
 
       Local<Function> filterFunc = filterValue.As<Function>();
 
-      context.filterCallback = [env,
-                                filterFunc](std::string_view item) -> bool {
-        // TODO(@jasnell): The use of ToLocalChecked here means that if
-        // the filter function throws an error the process will crash.
-        // The filterCallback should be updated to avoid the check and
-        // propagate the error correctly.
-        Local<Value> argv[] = {
-            String::NewFromUtf8(env->isolate(),
-                                item.data(),
-                                NewStringType::kNormal,
-                                static_cast<int>(item.size()))
-                .ToLocalChecked()};
-        Local<Value> result =
-            filterFunc->Call(env->context(), Null(env->isolate()), 1, argv)
-                .ToLocalChecked();
+      context.filterCallback =
+          [db, env, filterFunc](std::string_view item) -> bool {
+        Local<Value> argv[1];
+        if (!String::NewFromUtf8(env->isolate(),
+                                 item.data(),
+                                 NewStringType::kNormal,
+                                 static_cast<int>(item.size()))
```

Comment:
Suggestion: Use `ToV8Value()` here, which will actually throw a proper exception

(`String::NewFromUtf8()` _should_ throw an exception, but right now that's sadly just a TODO in the V8 source)